torch/lib/c10d/test/ProcessGroupGlooTest.cpp

```diff
@@ -347,6 +347,13 @@ void testRecv(const std::string& path) {
 
 TEST(ProcessGroupGlooTest, testSIGSTOPException) {
   // test SIGSTOP
+  // Fork() and TSAN don't play well together, so skip the test if we're testing
+  // with TSAN.
+  auto s = std::getenv("PYTORCH_TEST_WITH_TSAN");
+  if (s && strcmp(s, "1") == 0) {
+    return;
+  }
```

Comment:
Can we have a function in TestUtils.hpp that does this and just call it in both places here?

Reply:
Sounds good